shoulder/generator/__init__.py

```diff
@@ -54,6 +55,6 @@ def generate_all(objects, outdir):
             shutil.copy(config.accessor_macros_path, sub_outdir)
 
         g = generator_class()
-        g.generate(objects, sub_outdir)
+        g.generate(copy.deepcopy(regs), sub_outdir)
```

Comment:
> Is it so editing (filtering / transforming) them in one generator doesn't effect the other?

Yep, exactly. I was running into some unexpected behaviors when working on the C and C++ header generators at the same time, when one needs to use a transform and the other does not. For example, the C generator needs each field to have a unique name (using the unique_fieldset_names transform), while the C++ generator does not (it uses namespaces instead of unique function names)